### ISSUE TITLE
Initializer arguments not checked when raising exception with implicit instantiation.

### ISSUE THREAD

**Issue by emcd** [PyCQA/pylint#4033]
### Reproducer

```python
class Bang( Exception ):
    def __init__( self, required ):
        super( ).__init__( f"{required}" )

try: raise Bang
except TypeError as exc: print( exc )

try: raise Bang( )
except TypeError as exc: print( exc )
```

### Steps to reproduce
1. Run `pylint --score=no --reports=no --enable=all --disable=missing-module-docstring,missing-class-docstring,multiple-statements,too-few-public-methods` on the reproducer above.
2. Note that Pylint only reports about `no-value-for-parameter` for the attempt to explicitly instantiate the exception class before raising and not the attempt to raise the exception class directly (implicit instantiation - valid Python).
3. Run the reproducer in a Python interpreter and note that both argument mismatches (explicit instantiation and implicit instantiation) are caught correctly.

```
$ pylint --score=no --reports=no --enable=all --disable=missing-module-docstring,missing-class-docstring,multiple-statements,too-few-public-methods pylint-bugs/raise_class_exc_false_neg.py 
************* Module raise_class_exc_false_neg
pylint-bugs/raise_class_exc_false_neg.py   8,11 [no-value-for-parameter] No value for argument 'required' in constructor call
```
```
$ python3 pylint-bugs/raise_class_exc_false_neg.py 
__init__() missing 1 required positional argument: 'required'
__init__() missing 1 required positional argument: 'required'
$
```

### Current behavior
Pylint does not check arguments against parameters for exceptions raised with implicit instantiation.

### Expected behavior
Pylint checks arguments against parameters for exceptions raised with implicit instantiation.

### pylint --version output
```
$ pylint --versionpylint 2.6.0
astroid 2.4.2
Python 3.6.12 (default, Nov  8 2020, 15:40:05) 
[GCC 5.4.0 20160609]
```

**Comment by hippo91** [comment]
@emcd thanks for your report.
